extension/public/reddit.js

```diff
@@ -5,95 +5,116 @@ var Reddit = {
-    var h = '<div class="reddit-dashboard">';
-    h += '<div class="reddit-page-status">';
-    h +=
-      '<div class="reddit-status total_hidden"><span class="variable"></span>Total hidden</div>';
-    h += "</div>";
-    h += '<div class="reddit-controls">';
-    h += '<div class="reddit-control hide_links">Load Hide Buttons</div>';
-    h += '<div class="reddit-control unhide_links">Unhide all</div>';
-    h += "</div>";
-    h += "</div>";
-
-    return h;
+  // initalize
+  initialize: function () {
+      this.blurImage();
+      this.checkForUpdate();
   },
 
-  // initalize UI
-  initialize: function () {
-    this.setupUI();
+  //listener to scroll events
+  //updates new visible posts to be filtered
+  checkForUpdate: function(){
+      document.addEventListener('scroll', (ev) => {
+      this.counter++;
+      if(this.counter == this.refreshFrequency){
+          this.counter = 0;
+          this.blurImage();
+          //console.log("updated"); //debug
+      }
+      });
   },
 
-  // initalize all UI elements and onclick binding
-  setupUI: function () {
-    list = document.querySelector(this.listSelector);
-    list.innerHTML = this.dashboardHTML() + list.innerHTML;
 
-    document
-      .querySelector(".reddit-control.unhide_links")
-      .addEventListener("click", () => {
-        this.restoreDOM();
-        console.log("unhide all");
-      });
 
-    document
-      .querySelector(".reddit-control.hide_links")
-      .addEventListener("click", () => {
-        this.setupPostUI();
-        console.log("all links hidden");
-      });
+  //check if any configured keywords are found in a post
+  //return true if found
+  async filterText (post,callBack) {
+      //temporary filter data
+      //these data must be stored in cookie
+      const blockedWordsData = [
+        "trump",
+        "terrorist",
+        "communism",
+        "racism",
+        "isis",
+        "pizza"
+    ];
+
+      const title = this.getDOMTitle(post);
+      const description = this.getDOMDescription(post);
+      //image varible contain image url then contain result of ocr
+      //getDOMImageLink gets an url of the image if there is one
+      let image = this.getDOMImageLink(post);
+
+      //check if the imagelink was blank or not.
+      if(image != ""){
+        //asynchronously computing ocr, therefore await is needed
+        image = await this.ocr(image);
+        image = image.toLowerCase();
+        //callback to wait ocr to be processed
+        callBack();
+      }
+
+      //search if blockedwords are in the title
+      //true if found
+      var isInclude = false;
+      for (i in blockedWordsData){
+
+          if(title.includes(blockedWordsData[i])){
+            isInclude = true;
+          }else if (image.includes(blockedWordsData[i])){
+            isInclude = true;
+          }else if (description.includes(blockedWordsData[i])){
+            isInclude = true;
+          }
+      }
+      return isInclude;
```

Comment:
This detection code should probably run inside the callback right?

You're doing a detection for `image.includes()`, but image is always going to be the dom link as it'll run those checks before the async stuff has finished running.

Reply:
Been awhile since I've dealt with async/await in the JS environment though. @atan958 Is this faulty behavior or is this code fine?

Reply:
currently image is url link, so the image can be undefined.
async for this function is there to await ocr process. The image will contain the results of ocr so detection for image.includes() should be done after the ocr.
I think this is my thought to this function

Reply:
The way the callback has been done seems a little odd to me, but if you're confident that it works then I'm happy to merge it.